transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/EncodeTransformationPipelineBuilder.scala

```diff
@@ -0,0 +1,59 @@
+package org.broadinstitute.monster.encode.transformation
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.coders.Coder
+import java.time.OffsetDateTime
+import org.broadinstitute.monster.common.{PipelineBuilder, StorageIO}
+import org.broadinstitute.monster.common.msg._
+import org.broadinstitute.monster.encode.jadeschema.table.HumanDonor
+import upack.Msg
+
+object EncodeTransformationPipelineBuilder extends PipelineBuilder[Args] {
+  /** (De)serializer for the upack messages we read from storage. */
+  implicit val msgCoder: Coder[Msg] = Coder.beam(new UpackMsgCoder)
+
+  /**
+    * Schedule all the steps for the Encode transformation in the given pipeline context.
+    *
+    * Scheduled steps are launched against the context's runner when the `run()` method
+    * is called on it.
+    */
+  override def buildPipeline(ctx: ScioContext, args: Args): Unit = {
+    // read in extracted info
+    val donorInputs = StorageIO
+      .readJsonLists(
+        ctx,
+        "Donors",
+        s"${args.inputPrefix}/Donor/*.json"
+      )
+
+    val humanDonorOutput = donorInputs.map(transformDonor)
+
+    // write back to storage
+    StorageIO.writeJsonLists(
+      humanDonorOutput,
+      "HumanDonors",
+      s"${args.outputPrefix}/human_donor"
+    )
+    ()
+  }
+
+  def transformDonor(donorInput: Msg): HumanDonor = {
+    HumanDonor(
+      id = donorInput.read[String]("accession"),
+      crossReferences = donorInput.read[Array[String]]("dbxrefs"),
+      timeCreated = donorInput.read[OffsetDateTime]("date_created"),
+      age = donorInput.tryRead[String]("age"),
+      ethnicities = Array() ++ donorInput.tryRead[String]("ethnicity"),
```

Comment:
You can do this instead:
```suggestion
      ethnicities = donorInput.tryRead[Array[String]]("ethnicity").getOrElse(Array.empty),
```
Easier to understand the intent instead of the type-fu (at least to me)

Reply:
Wait... if there's only ever one ethnicity, why does the column need to be an array? (Array vs. not makes no difference for search in elasticsearch, but it does make SQL-ing more complicated in BigQuery)

Reply:
Yes, Encode is storing ethnicity as a single string but I believe other datasets store it as a list of strings. Since this falls into the realm of DSPCore, I think the idea is to choose a datatype that would work for multiple datasets.  @kreinold would you be able to chime in here and add anything I left out?

I believe the same would be said for the twin_ids field below.

Reply:
Agreed that it is good to align column names and data types (and even data values when possible!) of the DSP Core fields, but I don't think it's worth bending the data to align the BigQuery column modes. Arrays have extra storage overhead, and they're more difficult to work with in SQL. When we index for search we'll be targeting Elasticsearch where everything is an array.

Reply:
Oh, good to know! I hadn't realized that Elasticsearch stored things differently. I just updated it to store as an optional string instead of an array.